### DISCUSSION TITLE
Artifact LibreClinica-web:war: Error during artifact deployment.

### DISCUSSION THREAD

**Discussion by shanzhu2021** [reliatec-gmbh/LibreClinica#274] [Q&A]
I use a version downloaded this February. At that time, I succeeded in building and running it within my IDEA. During this period, I had removed my tomcat 8.5.61, and I unpacked this version of Tomcat distribution at the same place. But when I try to reproduce this process to run the LibreClinica, it persistently reports the error as above. I copy the complete error messages and attached here, please help me with this problem.

The message:
[deploy_output.TXT](https://github.com/reliatec-gmbh/LibreClinica/files/7199575/deploy_output.TXT)
The screenshot:
<img width="960" alt="deploy_screenshot" src="https://user-images.githubusercontent.com/77312250/134099919-c1d12a66-7d5f-4629-a293-da17c486668b.PNG">
Tomcat logs:
[logs.zip](https://github.com/reliatec-gmbh/LibreClinica/files/7199631/logs.zip)

**Comment by toskrip** [comment]
If I understand it correctly you have a new tomcat installation where you try to deploy the war archive. When I see your log file, I would say that you did not created libreclinica.config folder in new TOMCAT_HOME. Within libreclinica.config folder in TOMCAT_HOME there have to be datainfo.properties file copied and modified to reflect your DB connection.

see:

https://libreclinica-docs.readthedocs.io/en/latest/dev/dev-machine.html#debug

**Comment by shanzhu2021** [comment]
Thanks for your feedback and it seems the error disappear. But When I try to run it today a new problem pops up, details please refer to the following files:

The
[deploy_output_0924.TXT](https://github.com/reliatec-gmbh/LibreClinica/files/7226767/deploy_output_0924.TXT)
The screenshot:
<img width="960" alt="XJCError" src="https://user-images.githubusercontent.com/77312250/134699592-ce2b41e8-669c-420b-9062-dd7a1a0f3535.PNG">

**Reply by toskrip** [reply]
This is not longer deployment error, but a compilation error. Your log basically says that maven is executing XJC while building ODM module. XJC takes the XML schema files (.xsd) and creates annotated Java classes (model for XML marshalling unmarshalling, representation of ODM standard). Once it reach reference to W3C DigitalSignature schema (xmldsig-core-schema.xsd) it looks like it fails to parse it properly. I see some connection timeout errors in your log as well. I have not seen this type of issue before, my guess would be that XJC tries to check against W3C servers regarding the namespace "http://www.w3.org/2000/09/xmldsig#" and it cannot reach it...  maybe you do not have internet connection, or the communication is not allowed on network level.

**Reply by shanzhu2021** [reply]
When I make more attempt to get the system work, it reports this error repeatedly, but with different error messages:


[0930.TXT.txt](https://github.com/reliatec-gmbh/LibreClinica/files/7271368/0930.TXT.txt)

What is the problem and how to solve it?

**Reply by toskrip** [reply]
The error message that you are getting is still the same:

`org.xml.sax.SAXParseException: schema_reference.4: Failed to read schema document 'file:/C:/Users/Shanzhu/IdeaProjects/LibreClinica/odm/src/main/schemas/xmldsig-core-schema.xsd', because 1) could not find the document; 2) the document could not be read; 3) the root element of the document is not <xsd:schema>.`
